source/_posts/2019-03-11-upgrading-to-embark-4.md

```diff
@@ -0,0 +1,73 @@
+
+Thereby, we announce that you can now use the file pipeline that YOU want. Also, you can STILL use Embark's pipeline if you are fine with what is currently there.
+
+So, as you just read, it's not really a breaking change, well  that is, unless you decide you want to convert you Dapp from Embark's pipeline to the one of your choice.
+
+If you don't plan on starting a project with another pipeline or don't plan of converting, you can skip to [the next section](#New-Web3-plugin).
+
+### Converting to another pipeline
+
+The first thing you might want to do is change or add the new `generationDir` property in `embark.json`. This property tells Embark where to generation the various artifacts that you will need to build your Dapp. Most of those artifacts were already generated before, but were inside the `.embark/` folder. 
+
+Since most file pipelines need those files inside the source folder, we're going to put `generationDir` as your selected pipeline's source folder. For example,  create-react-app (CRA) has `src/` as source folder.
+
+Afterwards, we need to convert all "magic" imports to relative imports.
+
+The first one is the EmbarkJS import. The "magic" import is `"Embark/EmbarkJS"`. We need to convert that to the relative path. The `embarkjs.js` script is at the root of  `embarkArtifacts/`, so you need to convert it to something like `"./embarkArtifacts/embarkjs"`.
+
+Second ones are the contracts. They used to look like `"Embark/contract/ContractName"`. They are now located inside `embarkArtifactscontracts/`.
+
+Third one, there used to be `"Embark/web3"`, but we removed it. Don't worry, we are not removing web3 support, far from it. We actually just got rid of that useless import that didn't do much. All you have to do is use the global `web3` object that is avilable everywhere in the Dapp.
+
+Then, you can move your Embark files and configs inside a project created by the pipeline of your choice.
+
+### New project with another pipeline
+
+If you are starting your project from scratch, you have two options.
+
+The easier one is to use our new Embark CRA template (TODO add template URl). It sets up a simple Embark project for you with the all of the files you are used to. The only slight difference is that the config files are in `embarkConfig` to make sure they don't clash with CRA's config. It's already set up for you to start and use. You only need to do `embark run` and in another terminal `yarn start` or `npm run start` to start CRA.
+
+The other solution is to create a project with your chosen pipeline and then (TODO new command or we need to do it in another).
+
+## New Web3 plugin
+
+Starting with Embark 4 beta.2, Embark no longer comes with Web3js by default for your Dapp. Don't run. We did that so that we can now have the possibility of supporting more than just Web3js, but potentially EthersJS and more (even the possibility for you to do it).
```

Comment:
```suggestion
Starting with Embark 4 beta.2, Embark no longer supplies the Dapp with `Web3.js` by default. Don't run. We did that so that we can now have the possibility of supporting more than just `Web3.js`, such as EthersJS, and more. You can even roll your own.
```